stdlib/websub/src/main/ballerina/websub/h2_hub_persistence.bal

```diff
@@ -0,0 +1,175 @@
+                                            (topic VARCHAR(255), callback VARCHAR(255), secret VARCHAR(255),
+                                            lease_seconds BIGINT, created_at BIGINT, PRIMARY KEY (topic, callback))";
+const string INSERT_INTO_SUBSCRIPTIONS_TABLE = "INSERT INTO subscriptions
+                                                (topic,callback,secret,lease_seconds,created_at) VALUES (?,?,?,?,?)";
+const string DELETE_FROM_SUBSCRIPTIONS = "DELETE FROM subscriptions WHERE topic=? AND callback=?";
+const string SELECT_FROM_SUBSCRIPTIONS = "SELECT topic, callback, secret, lease_seconds, created_at FROM subscriptions";
+
+# Represents H2 based hub persistence configuration and functions.
+public type H2HubPersistenceObject object {
+
+    // TODO: make private
+    public h2:Client subscriptionDbClient;
+
+    public function __init(h2:Client subscriptionDbClient) {
+        self.subscriptionDbClient = subscriptionDbClient;
+        var ret = self.subscriptionDbClient->update(CREATE_TOPICS_TABLE);
+        if (ret is error) {
+            log:printError("'topics' table creation failed: " + <string>ret.detail().message);
+        }
+
+        ret = self.subscriptionDbClient->update(CREATE_SUBSCRIPTIONS_TABLE);
+        if (ret is error) {
+            log:printError("'subscriptions' table creation failed: " + <string>ret.detail().message);
+        }
+    }
+
+    # Function to add or update subscription details.
+    #
+    # + subscriptionDetails - The details of the subscription to add or update
+    public function addSubscription(SubscriptionDetails subscriptionDetails) {
+        sql:Parameter para1 = { sqlType: sql:TYPE_VARCHAR, value: subscriptionDetails.topic };
+        sql:Parameter para2 = { sqlType: sql:TYPE_VARCHAR, value: subscriptionDetails.callback };
+        sql:Parameter para3 = { sqlType: sql:TYPE_VARCHAR, value: subscriptionDetails.secret };
+        sql:Parameter para4 = { sqlType: sql:TYPE_BIGINT, value: subscriptionDetails.leaseSeconds };
+        sql:Parameter para5 = { sqlType: sql:TYPE_BIGINT, value: subscriptionDetails.createdAt };
+
+        var rowCount = self.subscriptionDbClient->update(DELETE_FROM_SUBSCRIPTIONS, untaint para1, untaint para2);
+        if (rowCount is int) {
+            log:printDebug("Successfully removed " + rowCount + " entries for existing subscription");
+        } else if (rowCount is error) {
+            string errCause = <string> rowCount.detail().message;
+            log:printError("Error occurred deleting subscription data: " + errCause);
+        }
+
+        rowCount = self.subscriptionDbClient->update(INSERT_INTO_SUBSCRIPTIONS_TABLE, untaint para1, untaint para2,
+                                                    untaint para3, untaint para4, untaint para5);
+        if (rowCount is int) {
+            log:printDebug("Successfully updated " + rowCount + " entries for subscription");
+        } else if (rowCount is error) {
```

Comment:
The return type of update remote function is `int|error` so we could simply have an `else` block instead of an `else if`. This applies to several similar places.